### DISCUSSION TITLE
Too many connections will cause axum server to break down

### DISCUSSION THREAD

**Discussion by gy0801151351** [tokio-rs/axum#2388] [Q&A]
When I use a few test clients by "ab" command within "httpd" tool, too many connections will cause axum server to break down.

### Version
axum v0.7.1

### Platform

ubuntu 20

### Crates

axum 0.7.1

### Description
You can test by using any of the examples.

**Comment by davidpdrsn** [comment]
Please be more specific. "Breaks down" how?

I doubt it's an issue within axum itself. Are you able to reproduce it using hyper directly, without axum?

**Reply by gy0801151351** [reply]
I think this is not an error about hyper. Because  user should process tcp connections by self.
Belows are my test steps.

1. Run a demo axum server
2. Use "ab" command of "httpd" tool to begin stress test. set requests over 1000000 or more, set concurrency 2000 or more.
3. The axum server will break down and throw "Too many open files" error.

**Reply by gy0801151351** [reply]
The version "0.6.20" of axum crate dealed with the connection error. But the version "0.7.1" doesn't.

This is the code snapshot of version "0.7.1"
![图片](https://github.com/tokio-rs/axum/assets/16450354/55574c96-c526-4b5c-9da6-e543814394c1)

This is the code snapshot of version "0.6.20"
![图片](https://github.com/tokio-rs/axum/assets/16450354/457fd080-58ab-45aa-a4fc-23fa93f648e6)

**Reply by davidpdrsn** [reply]
Ah yeah its entirely possible there is some additional error handling to do in `serve` that hyper used to do but now doesn't because `hyper::Server` was removed!

**Reply by davidpdrsn** [reply]
Should be fixed in https://github.com/tokio-rs/axum/pull/2400.

Thanks for noticing!

**Comment by dayvejones** [comment]
Hello, @gy0801151351 !

Can you also measure the metrics (cpu usage, ram, rps, size of compiled files) for your experimental programs on 0.6.20 and 0.7.1 on the same server? For different load levels when the axum server does not crash.

https://github.com/tokio-rs/axum/discussions/2381 - My comparison results were a little disappointing. A little later I will deal with the profiler.

I would be grateful if you could share your results too.

**Reply by gy0801151351** [reply]
On my computer, when I open connections over 1000, the axum(v0.7.1) server will break down. So I have no metrics to get.
